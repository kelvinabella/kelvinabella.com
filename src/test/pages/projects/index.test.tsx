import React from "react";
import { render, RouterCntxt } from "~/test/utils/index";
import Projects, { getStaticProps } from "~/pages/projects";

jest.mock("~/helpers/api", () => ({
  getLatestProjects: () =>
    Promise.resolve([
      {
        name: "kelvinabella.com",
        githubUrl: "https://kelvinabella.com",
        homepage: "https://kelvinabella.com",
        technology: "react",
        description: "description",
      },
    ]),
}));

// https://medium.com/@rickhanlonii/understanding-jest-mocks-f0046c68e53c
describe("Projects page", () => {
  it("renders Projects correctly", async () => {
    const { asFragment } = render(
      <RouterCntxt>
        <Projects latestProjects={[]} />
      </RouterCntxt>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders Projects correctly", async () => {
    const { asFragment } = render(
      <RouterCntxt>
        <Projects
          latestProjects={[
            {
              name: "kelvinabella.com",
              githubUrl: "https://kelvinabella.com",
              homepage: "https://kelvinabella.com",
              technology: "react",
              description: "description",
            },
          ]}
        />
      </RouterCntxt>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  // https://stackoverflow.com/questions/58490318/how-do-you-write-jest-tests-for-getinitialprops
  it("populates the latestProjects from API", async () => {
    const { props } = await getStaticProps();

    expect(props).toEqual({
      latestProjects: [
        {
          name: "kelvinabella.com",
          githubUrl: "https://kelvinabella.com",
          homepage: "https://kelvinabella.com",
          technology: "react",
          description: "description",
        },
      ],
    });
  });
});
