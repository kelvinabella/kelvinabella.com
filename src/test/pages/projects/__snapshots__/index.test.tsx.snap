// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects page renders Projects correctly 1`] = `
<DocumentFragment>
  <header
    class="style__NavContainer-l9pfiu-0 fLqkdz"
  >
    <nav
      class="style__MenuWrapper-l9pfiu-1 jvDcQk"
    >
      <div
        class="style__LogoContainer-sc-10h1uvs-0 iwUkZR"
      >
        <a
          aria-label="Kelvs"
          href="/"
        >
          <svg
            class="logo"
            fill="none"
            height="218"
            viewBox="0 0 850 218"
            width="850"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Kelvs
            </title>
            <path
              d="M84.839 172.349H87.339V169.849V135.46L116.028 112.319L163.302 171.411L164.052 172.349H165.254H190.024H195.278L191.963 168.272L134.409 97.4748L193.342 49.6862L198.82 45.2444H191.768H164.556H163.663L162.972 45.8098L113.97 85.9188L113.956 85.9299L87.339 107.405V47.7444V45.2444H84.839H66H63.5V47.7444V169.849V172.349H66H84.839ZM240.946 45.2444H238.446V47.7444V169.849V172.349H240.946H337.409H339.909V169.849V151.708V149.208H337.409H262.285V118.797H327.815H330.315V116.297V98.1562V95.6562H327.815H262.285V68.3856H337.409H339.909V65.8856V47.7444V45.2444H337.409H240.946ZM489.283 151.708V149.208H486.783H412.531V47.7444V45.2444H410.031H391.192H388.692V47.7444V169.849V172.349H391.192H486.783H489.283V169.849V151.708ZM641.823 48.7512L643.367 45.2444H639.535H617.382H615.693L615.063 46.8111L585.583 120.074L585.576 120.092L585.569 120.11C583.842 124.599 581.824 129.444 579.513 134.644L579.503 134.666L579.494 134.687C578.132 137.914 576.813 141.075 575.538 144.171C574.209 141.008 572.748 137.76 571.154 134.427C568.724 129.103 566.586 124.249 564.739 119.863L564.732 119.847L564.726 119.832L532.804 46.7437L532.149 45.2444H530.513H509.058H505.191L506.778 48.7703L561.725 170.875L562.388 172.349H564.005H585.809H587.44L588.097 170.856L641.823 48.7512ZM685.264 130.671L683.277 128.467L681.442 130.8L669.232 146.325L667.939 147.968L669.328 149.531C677.013 158.177 685.885 164.468 695.939 168.344C705.92 172.192 717.046 174.093 729.284 174.093C740.673 174.093 750.644 172.493 759.14 169.216L759.148 169.213L759.157 169.209C767.746 165.822 774.492 161.041 779.242 154.793C784.129 148.4 786.556 140.94 786.556 132.52C786.556 126.19 785.003 120.752 781.714 116.393C778.652 112.067 774.474 108.676 769.266 106.195C764.299 103.773 758.649 101.857 752.338 100.428C746.096 99.0144 739.569 97.8971 732.759 97.074C725.863 96.0385 719.864 94.8362 714.752 93.4729C709.747 92.1384 706.06 90.3562 703.527 88.2364C701.221 86.1968 700.036 83.5376 700.036 80.0149C700.036 77.764 700.637 75.8633 701.8 74.2209C702.982 72.5527 704.838 71.0216 707.533 69.7004L707.551 69.6917L707.568 69.6828C713.051 66.8875 720.018 65.4202 728.586 65.4202C736.983 65.4202 744.192 67.0422 750.287 70.1988C756.512 73.4225 761.641 77.5681 765.711 82.6284L767.529 84.8884L769.496 82.7572L782.056 69.1512L783.433 67.6595L782.255 66.0057C777.073 58.7252 769.884 53.163 760.796 49.2845C751.807 45.3988 741.222 43.5 729.109 43.5C719.321 43.5 710.306 44.992 702.091 48.0108C693.973 50.9202 687.374 55.2055 682.401 60.9215C677.315 66.6481 674.802 73.6462 674.802 81.7592C674.802 89.453 676.923 96.0798 681.29 101.474L681.299 101.485L681.308 101.496C685.586 106.655 691.545 110.708 699.044 113.732L699.066 113.741L699.088 113.749C706.629 116.622 715.423 118.754 725.445 120.168L725.458 120.17L725.471 120.172C733.447 121.212 740.229 122.363 745.831 123.621L745.854 123.626L745.878 123.631C751.403 124.759 755.291 126.344 757.774 128.231L757.792 128.245L757.811 128.259C760.172 129.984 761.322 132.248 761.322 135.311C761.322 138.995 759.958 141.969 757.133 144.405C754.227 146.895 750.347 148.841 745.385 150.164C740.369 151.502 735.178 152.173 729.807 152.173C720.01 152.173 711.774 150.317 705.005 146.717C698.081 142.919 691.498 137.585 685.264 130.671Z"
              fill="currentColor"
              stroke="currentColor"
              stroke-width="5"
            />
            <rect
              height="213"
              rx="17.5"
              stroke="currentColor"
              stroke-width="5"
              width="845"
              x="2.5"
              y="2.5"
            />
          </svg>
        </a>
      </div>
      <div
        class="style__MenuDesktopContainer-ymbyh2-0 crRbTX"
      >
        <span
          class="style__LinkWrapper-ymbyh2-1 fAKByD"
        >
          <a
            href="/about"
          >
            About
          </a>
        </span>
        <span
          class="style__LinkWrapper-ymbyh2-1 fAKByD"
        >
          <a
            href="/blog"
          >
            Blog
          </a>
        </span>
        <span
          class="style__LinkWrapper-ymbyh2-1 fAKByD"
        >
          <a
            href="/projects"
          >
            Projects
          </a>
        </span>
        <span
          class="style__LinkWrapper-ymbyh2-1 fAKByD"
        >
          <a
            href="/contact"
          >
            Contact
          </a>
        </span>
        <span
          class="style__LinkWrapper-ymbyh2-1 fAKByD"
        >
          <a
            href="/ama"
          >
            AMA
          </a>
        </span>
      </div>
      <div
        class="style__IconContainer-sc-1kqqcq8-0 fgJKFR"
      >
        <span
          aria-label="dark mode"
          class="style__AnimatedSpan-sc-1kqqcq8-1 dMwwia"
          role="button"
          style="transform: rotate(90deg);"
          tabindex="0"
        >
          <svg
            class="sun"
            height="480pt"
            viewBox="0 0 480 480"
            width="480pt"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m368 240c0 70.691406-57.308594 128-128 128s-128-57.308594-128-128 57.308594-128 128-128 128 57.308594 128 128zm0 0"
            />
            <path
              d="m248 472v-64c0-4.417969-3.582031-8-8-8s-8 3.582031-8 8v64c0 4.417969 3.582031 8 8 8s8-3.582031 8-8zm0 0"
            />
            <path
              d="m240 80c4.417969 0 8-3.582031 8-8v-64c0-4.417969-3.582031-8-8-8s-8 3.582031-8 8v64c0 4.417969 3.582031 8 8 8zm0 0"
            />
            <path
              d="m114.34375 125.65625c3.140625 3.03125 8.128906 2.988281 11.214844-.097656 3.085937-3.085938 3.128906-8.074219.097656-11.214844l-48-48c-3.140625-3.03125-8.128906-2.988281-11.214844.097656-3.085937 3.085938-3.128906 8.074219-.097656 11.214844zm0 0"
            />
            <path
              d="m365.65625 354.34375c-3.140625-3.03125-8.128906-2.988281-11.214844.097656-3.085937 3.085938-3.128906 8.074219-.097656 11.214844l48 48c3.140625 3.03125 8.128906 2.988281 11.214844-.097656 3.085937-3.085938 3.128906-8.074219.097656-11.214844zm0 0"
            />
            <path
              d="m360 128c2.121094 0 4.15625-.84375 5.65625-2.34375l48-48c3.03125-3.140625 2.988281-8.128906-.097656-11.214844-3.085938-3.085937-8.074219-3.128906-11.214844-.097656l-48 48c-2.285156 2.289062-2.972656 5.730469-1.734375 8.71875s4.15625 4.9375 7.390625 4.9375zm0 0"
            />
            <path
              d="m472 232h-64c-4.417969 0-8 3.582031-8 8s3.582031 8 8 8h64c4.417969 0 8-3.582031 8-8s-3.582031-8-8-8zm0 0"
            />
            <path
              d="m8 248h64c4.417969 0 8-3.582031 8-8s-3.582031-8-8-8h-64c-4.417969 0-8 3.582031-8 8s3.582031 8 8 8zm0 0"
            />
            <path
              d="m77.65625 413.65625 48-48c3.03125-3.140625 2.988281-8.128906-.097656-11.214844-3.085938-3.085937-8.074219-3.128906-11.214844-.097656l-48 48c-2.078125 2.007812-2.914062 4.984375-2.179688 7.78125.730469 2.796875 2.914063 4.980469 5.710938 5.710938 2.796875.734374 5.773438-.101563 7.78125-2.179688zm0 0"
            />
          </svg>
        </span>
      </div>
      <div
        class="style__MenuMobileContainer-sc-1d9cvo5-0 dWlKSS"
      >
        <div
          class="style__LinkWrapper-sc-1d9cvo5-1 biJFEi"
        >
          <span
            style="transform: translateX(-150px);"
          >
            <a
              href="/about"
            >
              About
            </a>
          </span>
          <span
            style="transform: translateX(-150px);"
          >
            <a
              href="/blog"
            >
              Blog
            </a>
          </span>
          <span
            style="transform: translateX(-150px);"
          >
            <a
              href="/projects"
            >
              Projects
            </a>
          </span>
          <span
            style="transform: translateX(-150px);"
          >
            <a
              href="/contact"
            >
              Contact
            </a>
          </span>
          <span
            style="transform: translateX(-150px);"
          >
            <a
              href="/ama"
            >
              AMA
            </a>
          </span>
        </div>
        <div
          class="style__IconContainer-sc-1kqqcq8-0 fgJKFR"
        >
          <span
            aria-label="dark mode"
            class="style__AnimatedSpan-sc-1kqqcq8-1 dMwwia"
            role="button"
            style="transform: rotate(90deg);"
            tabindex="0"
          >
            <svg
              class="sun"
              height="480pt"
              viewBox="0 0 480 480"
              width="480pt"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m368 240c0 70.691406-57.308594 128-128 128s-128-57.308594-128-128 57.308594-128 128-128 128 57.308594 128 128zm0 0"
              />
              <path
                d="m248 472v-64c0-4.417969-3.582031-8-8-8s-8 3.582031-8 8v64c0 4.417969 3.582031 8 8 8s8-3.582031 8-8zm0 0"
              />
              <path
                d="m240 80c4.417969 0 8-3.582031 8-8v-64c0-4.417969-3.582031-8-8-8s-8 3.582031-8 8v64c0 4.417969 3.582031 8 8 8zm0 0"
              />
              <path
                d="m114.34375 125.65625c3.140625 3.03125 8.128906 2.988281 11.214844-.097656 3.085937-3.085938 3.128906-8.074219.097656-11.214844l-48-48c-3.140625-3.03125-8.128906-2.988281-11.214844.097656-3.085937 3.085938-3.128906 8.074219-.097656 11.214844zm0 0"
              />
              <path
                d="m365.65625 354.34375c-3.140625-3.03125-8.128906-2.988281-11.214844.097656-3.085937 3.085938-3.128906 8.074219-.097656 11.214844l48 48c3.140625 3.03125 8.128906 2.988281 11.214844-.097656 3.085937-3.085938 3.128906-8.074219.097656-11.214844zm0 0"
              />
              <path
                d="m360 128c2.121094 0 4.15625-.84375 5.65625-2.34375l48-48c3.03125-3.140625 2.988281-8.128906-.097656-11.214844-3.085938-3.085937-8.074219-3.128906-11.214844-.097656l-48 48c-2.285156 2.289062-2.972656 5.730469-1.734375 8.71875s4.15625 4.9375 7.390625 4.9375zm0 0"
              />
              <path
                d="m472 232h-64c-4.417969 0-8 3.582031-8 8s3.582031 8 8 8h64c4.417969 0 8-3.582031 8-8s-3.582031-8-8-8zm0 0"
              />
              <path
                d="m8 248h64c4.417969 0 8-3.582031 8-8s-3.582031-8-8-8h-64c-4.417969 0-8 3.582031-8 8s3.582031 8 8 8zm0 0"
              />
              <path
                d="m77.65625 413.65625 48-48c3.03125-3.140625 2.988281-8.128906-.097656-11.214844-3.085938-3.085937-8.074219-3.128906-11.214844-.097656l-48 48c-2.078125 2.007812-2.914062 4.984375-2.179688 7.78125.730469 2.796875 2.914063 4.980469 5.710938 5.710938 2.796875.734374 5.773438-.101563 7.78125-2.179688zm0 0"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        class="style__HamburgerWrapper-sc-1d9cvo5-2 ekkuZo"
      >
        <svg
          height="35"
          viewBox="10 10 70 70"
          width="35"
        >
          <title>
            Hamburger Menu
          </title>
          <path
            class="line top"
            d="m 30,33 h 40 c 13.100415,0 14.380204,31.80258 6.899646,33.421777 -24.612039,5.327373 9.016154,-52.337577 -12.75751,-30.563913 l -28.284272,28.284272"
          />
          <path
            class="line middle"
            d="m 70,50 c 0,0 -32.213436,0 -40,0 -7.786564,0 -6.428571,-4.640244 -6.428571,-8.571429 0,-5.895471 6.073743,-11.783399 12.286435,-5.570707 6.212692,6.212692 28.284272,28.284272 28.284272,28.284272"
          />
          <path
            class="line bottom"
            d="m 69.575405,67.073826 h -40 c -13.100415,0 -14.380204,-31.80258 -6.899646,-33.421777 24.612039,-5.327373 -9.016154,52.337577 12.75751,30.563913 l 28.284272,-28.284272"
          />
        </svg>
      </div>
    </nav>
  </header>
  <section
    class="style__Container-sc-5ez25-0 gqQPmb"
  >
    <div
      class="style__Apps-xfidxt-0 lUPEw"
    />
  </section>
  <footer
    class="style__FooterContainer-sc-1egeub5-0 jrWLre"
  >
    <div>
      <span>
        powered by 
        <a
          href="https://nextjs.org"
        >
          Next.js
        </a>
         and 
        <a
          href="https://reactjs.org"
        >
          React
        </a>
      </span>
      <span>
         | built by 
        <a
          href="https://kelvinabella.com"
        >
          Kelvin Abella
        </a>
         with ♥
      </span>
    </div>
  </footer>
</DocumentFragment>
`;

exports[`Projects page renders Projects correctly 2`] = `
<DocumentFragment>
  <header
    class="style__NavContainer-l9pfiu-0 fLqkdz"
  >
    <nav
      class="style__MenuWrapper-l9pfiu-1 jvDcQk"
    >
      <div
        class="style__LogoContainer-sc-10h1uvs-0 iwUkZR"
      >
        <a
          aria-label="Kelvs"
          href="/"
        >
          <svg
            class="logo"
            fill="none"
            height="218"
            viewBox="0 0 850 218"
            width="850"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Kelvs
            </title>
            <path
              d="M84.839 172.349H87.339V169.849V135.46L116.028 112.319L163.302 171.411L164.052 172.349H165.254H190.024H195.278L191.963 168.272L134.409 97.4748L193.342 49.6862L198.82 45.2444H191.768H164.556H163.663L162.972 45.8098L113.97 85.9188L113.956 85.9299L87.339 107.405V47.7444V45.2444H84.839H66H63.5V47.7444V169.849V172.349H66H84.839ZM240.946 45.2444H238.446V47.7444V169.849V172.349H240.946H337.409H339.909V169.849V151.708V149.208H337.409H262.285V118.797H327.815H330.315V116.297V98.1562V95.6562H327.815H262.285V68.3856H337.409H339.909V65.8856V47.7444V45.2444H337.409H240.946ZM489.283 151.708V149.208H486.783H412.531V47.7444V45.2444H410.031H391.192H388.692V47.7444V169.849V172.349H391.192H486.783H489.283V169.849V151.708ZM641.823 48.7512L643.367 45.2444H639.535H617.382H615.693L615.063 46.8111L585.583 120.074L585.576 120.092L585.569 120.11C583.842 124.599 581.824 129.444 579.513 134.644L579.503 134.666L579.494 134.687C578.132 137.914 576.813 141.075 575.538 144.171C574.209 141.008 572.748 137.76 571.154 134.427C568.724 129.103 566.586 124.249 564.739 119.863L564.732 119.847L564.726 119.832L532.804 46.7437L532.149 45.2444H530.513H509.058H505.191L506.778 48.7703L561.725 170.875L562.388 172.349H564.005H585.809H587.44L588.097 170.856L641.823 48.7512ZM685.264 130.671L683.277 128.467L681.442 130.8L669.232 146.325L667.939 147.968L669.328 149.531C677.013 158.177 685.885 164.468 695.939 168.344C705.92 172.192 717.046 174.093 729.284 174.093C740.673 174.093 750.644 172.493 759.14 169.216L759.148 169.213L759.157 169.209C767.746 165.822 774.492 161.041 779.242 154.793C784.129 148.4 786.556 140.94 786.556 132.52C786.556 126.19 785.003 120.752 781.714 116.393C778.652 112.067 774.474 108.676 769.266 106.195C764.299 103.773 758.649 101.857 752.338 100.428C746.096 99.0144 739.569 97.8971 732.759 97.074C725.863 96.0385 719.864 94.8362 714.752 93.4729C709.747 92.1384 706.06 90.3562 703.527 88.2364C701.221 86.1968 700.036 83.5376 700.036 80.0149C700.036 77.764 700.637 75.8633 701.8 74.2209C702.982 72.5527 704.838 71.0216 707.533 69.7004L707.551 69.6917L707.568 69.6828C713.051 66.8875 720.018 65.4202 728.586 65.4202C736.983 65.4202 744.192 67.0422 750.287 70.1988C756.512 73.4225 761.641 77.5681 765.711 82.6284L767.529 84.8884L769.496 82.7572L782.056 69.1512L783.433 67.6595L782.255 66.0057C777.073 58.7252 769.884 53.163 760.796 49.2845C751.807 45.3988 741.222 43.5 729.109 43.5C719.321 43.5 710.306 44.992 702.091 48.0108C693.973 50.9202 687.374 55.2055 682.401 60.9215C677.315 66.6481 674.802 73.6462 674.802 81.7592C674.802 89.453 676.923 96.0798 681.29 101.474L681.299 101.485L681.308 101.496C685.586 106.655 691.545 110.708 699.044 113.732L699.066 113.741L699.088 113.749C706.629 116.622 715.423 118.754 725.445 120.168L725.458 120.17L725.471 120.172C733.447 121.212 740.229 122.363 745.831 123.621L745.854 123.626L745.878 123.631C751.403 124.759 755.291 126.344 757.774 128.231L757.792 128.245L757.811 128.259C760.172 129.984 761.322 132.248 761.322 135.311C761.322 138.995 759.958 141.969 757.133 144.405C754.227 146.895 750.347 148.841 745.385 150.164C740.369 151.502 735.178 152.173 729.807 152.173C720.01 152.173 711.774 150.317 705.005 146.717C698.081 142.919 691.498 137.585 685.264 130.671Z"
              fill="currentColor"
              stroke="currentColor"
              stroke-width="5"
            />
            <rect
              height="213"
              rx="17.5"
              stroke="currentColor"
              stroke-width="5"
              width="845"
              x="2.5"
              y="2.5"
            />
          </svg>
        </a>
      </div>
      <div
        class="style__MenuDesktopContainer-ymbyh2-0 crRbTX"
      >
        <span
          class="style__LinkWrapper-ymbyh2-1 fAKByD"
        >
          <a
            href="/about"
          >
            About
          </a>
        </span>
        <span
          class="style__LinkWrapper-ymbyh2-1 fAKByD"
        >
          <a
            href="/blog"
          >
            Blog
          </a>
        </span>
        <span
          class="style__LinkWrapper-ymbyh2-1 fAKByD"
        >
          <a
            href="/projects"
          >
            Projects
          </a>
        </span>
        <span
          class="style__LinkWrapper-ymbyh2-1 fAKByD"
        >
          <a
            href="/contact"
          >
            Contact
          </a>
        </span>
        <span
          class="style__LinkWrapper-ymbyh2-1 fAKByD"
        >
          <a
            href="/ama"
          >
            AMA
          </a>
        </span>
      </div>
      <div
        class="style__IconContainer-sc-1kqqcq8-0 fgJKFR"
      >
        <span
          aria-label="dark mode"
          class="style__AnimatedSpan-sc-1kqqcq8-1 dMwwia"
          role="button"
          style="transform: rotate(90deg);"
          tabindex="0"
        >
          <svg
            class="sun"
            height="480pt"
            viewBox="0 0 480 480"
            width="480pt"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m368 240c0 70.691406-57.308594 128-128 128s-128-57.308594-128-128 57.308594-128 128-128 128 57.308594 128 128zm0 0"
            />
            <path
              d="m248 472v-64c0-4.417969-3.582031-8-8-8s-8 3.582031-8 8v64c0 4.417969 3.582031 8 8 8s8-3.582031 8-8zm0 0"
            />
            <path
              d="m240 80c4.417969 0 8-3.582031 8-8v-64c0-4.417969-3.582031-8-8-8s-8 3.582031-8 8v64c0 4.417969 3.582031 8 8 8zm0 0"
            />
            <path
              d="m114.34375 125.65625c3.140625 3.03125 8.128906 2.988281 11.214844-.097656 3.085937-3.085938 3.128906-8.074219.097656-11.214844l-48-48c-3.140625-3.03125-8.128906-2.988281-11.214844.097656-3.085937 3.085938-3.128906 8.074219-.097656 11.214844zm0 0"
            />
            <path
              d="m365.65625 354.34375c-3.140625-3.03125-8.128906-2.988281-11.214844.097656-3.085937 3.085938-3.128906 8.074219-.097656 11.214844l48 48c3.140625 3.03125 8.128906 2.988281 11.214844-.097656 3.085937-3.085938 3.128906-8.074219.097656-11.214844zm0 0"
            />
            <path
              d="m360 128c2.121094 0 4.15625-.84375 5.65625-2.34375l48-48c3.03125-3.140625 2.988281-8.128906-.097656-11.214844-3.085938-3.085937-8.074219-3.128906-11.214844-.097656l-48 48c-2.285156 2.289062-2.972656 5.730469-1.734375 8.71875s4.15625 4.9375 7.390625 4.9375zm0 0"
            />
            <path
              d="m472 232h-64c-4.417969 0-8 3.582031-8 8s3.582031 8 8 8h64c4.417969 0 8-3.582031 8-8s-3.582031-8-8-8zm0 0"
            />
            <path
              d="m8 248h64c4.417969 0 8-3.582031 8-8s-3.582031-8-8-8h-64c-4.417969 0-8 3.582031-8 8s3.582031 8 8 8zm0 0"
            />
            <path
              d="m77.65625 413.65625 48-48c3.03125-3.140625 2.988281-8.128906-.097656-11.214844-3.085938-3.085937-8.074219-3.128906-11.214844-.097656l-48 48c-2.078125 2.007812-2.914062 4.984375-2.179688 7.78125.730469 2.796875 2.914063 4.980469 5.710938 5.710938 2.796875.734374 5.773438-.101563 7.78125-2.179688zm0 0"
            />
          </svg>
        </span>
      </div>
      <div
        class="style__MenuMobileContainer-sc-1d9cvo5-0 dWlKSS"
      >
        <div
          class="style__LinkWrapper-sc-1d9cvo5-1 biJFEi"
        >
          <span
            style="transform: translateX(-150px);"
          >
            <a
              href="/about"
            >
              About
            </a>
          </span>
          <span
            style="transform: translateX(-150px);"
          >
            <a
              href="/blog"
            >
              Blog
            </a>
          </span>
          <span
            style="transform: translateX(-150px);"
          >
            <a
              href="/projects"
            >
              Projects
            </a>
          </span>
          <span
            style="transform: translateX(-150px);"
          >
            <a
              href="/contact"
            >
              Contact
            </a>
          </span>
          <span
            style="transform: translateX(-150px);"
          >
            <a
              href="/ama"
            >
              AMA
            </a>
          </span>
        </div>
        <div
          class="style__IconContainer-sc-1kqqcq8-0 fgJKFR"
        >
          <span
            aria-label="dark mode"
            class="style__AnimatedSpan-sc-1kqqcq8-1 dMwwia"
            role="button"
            style="transform: rotate(90deg);"
            tabindex="0"
          >
            <svg
              class="sun"
              height="480pt"
              viewBox="0 0 480 480"
              width="480pt"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m368 240c0 70.691406-57.308594 128-128 128s-128-57.308594-128-128 57.308594-128 128-128 128 57.308594 128 128zm0 0"
              />
              <path
                d="m248 472v-64c0-4.417969-3.582031-8-8-8s-8 3.582031-8 8v64c0 4.417969 3.582031 8 8 8s8-3.582031 8-8zm0 0"
              />
              <path
                d="m240 80c4.417969 0 8-3.582031 8-8v-64c0-4.417969-3.582031-8-8-8s-8 3.582031-8 8v64c0 4.417969 3.582031 8 8 8zm0 0"
              />
              <path
                d="m114.34375 125.65625c3.140625 3.03125 8.128906 2.988281 11.214844-.097656 3.085937-3.085938 3.128906-8.074219.097656-11.214844l-48-48c-3.140625-3.03125-8.128906-2.988281-11.214844.097656-3.085937 3.085938-3.128906 8.074219-.097656 11.214844zm0 0"
              />
              <path
                d="m365.65625 354.34375c-3.140625-3.03125-8.128906-2.988281-11.214844.097656-3.085937 3.085938-3.128906 8.074219-.097656 11.214844l48 48c3.140625 3.03125 8.128906 2.988281 11.214844-.097656 3.085937-3.085938 3.128906-8.074219.097656-11.214844zm0 0"
              />
              <path
                d="m360 128c2.121094 0 4.15625-.84375 5.65625-2.34375l48-48c3.03125-3.140625 2.988281-8.128906-.097656-11.214844-3.085938-3.085937-8.074219-3.128906-11.214844-.097656l-48 48c-2.285156 2.289062-2.972656 5.730469-1.734375 8.71875s4.15625 4.9375 7.390625 4.9375zm0 0"
              />
              <path
                d="m472 232h-64c-4.417969 0-8 3.582031-8 8s3.582031 8 8 8h64c4.417969 0 8-3.582031 8-8s-3.582031-8-8-8zm0 0"
              />
              <path
                d="m8 248h64c4.417969 0 8-3.582031 8-8s-3.582031-8-8-8h-64c-4.417969 0-8 3.582031-8 8s3.582031 8 8 8zm0 0"
              />
              <path
                d="m77.65625 413.65625 48-48c3.03125-3.140625 2.988281-8.128906-.097656-11.214844-3.085938-3.085937-8.074219-3.128906-11.214844-.097656l-48 48c-2.078125 2.007812-2.914062 4.984375-2.179688 7.78125.730469 2.796875 2.914063 4.980469 5.710938 5.710938 2.796875.734374 5.773438-.101563 7.78125-2.179688zm0 0"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        class="style__HamburgerWrapper-sc-1d9cvo5-2 ekkuZo"
      >
        <svg
          height="35"
          viewBox="10 10 70 70"
          width="35"
        >
          <title>
            Hamburger Menu
          </title>
          <path
            class="line top"
            d="m 30,33 h 40 c 13.100415,0 14.380204,31.80258 6.899646,33.421777 -24.612039,5.327373 9.016154,-52.337577 -12.75751,-30.563913 l -28.284272,28.284272"
          />
          <path
            class="line middle"
            d="m 70,50 c 0,0 -32.213436,0 -40,0 -7.786564,0 -6.428571,-4.640244 -6.428571,-8.571429 0,-5.895471 6.073743,-11.783399 12.286435,-5.570707 6.212692,6.212692 28.284272,28.284272 28.284272,28.284272"
          />
          <path
            class="line bottom"
            d="m 69.575405,67.073826 h -40 c -13.100415,0 -14.380204,-31.80258 -6.899646,-33.421777 24.612039,-5.327373 -9.016154,52.337577 12.75751,30.563913 l 28.284272,-28.284272"
          />
        </svg>
      </div>
    </nav>
  </header>
  <section
    class="style__Container-sc-5ez25-0 gqQPmb"
  >
    <div
      class="style__Apps-xfidxt-0 lUPEw"
    >
      <div
        class="style__AppCard-xfidxt-1 hraBKP"
      >
        <div
          class="style__AppLinks-xfidxt-3 eCCOvu"
        >
          <div
            class="style__HorizontalBar-sc-5ez25-1 jczapz"
          />
          <span>
            <a
              aria-label="Open Link"
              href="https://kelvinabella.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                role="img"
                viewBox="0 0 194.818 194.818"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  External
                </title>
                <g>
                  <path
                    d="M185.818,2.161h-57.04c-4.971,0-9,4.029-9,9s4.029,9,9,9h35.312l-86.3,86.3c-3.515,3.515-3.515,9.213,0,12.728 c1.758,1.757,4.061,2.636,6.364,2.636s4.606-0.879,6.364-2.636l86.3-86.3v35.313c0,4.971,4.029,9,9,9s9-4.029,9-9v-57.04 C194.818,6.19,190.789,2.161,185.818,2.161z"
                  />
                  <path
                    d="M149,77.201c-4.971,0-9,4.029-9,9v88.456H18v-122h93.778c4.971,0,9-4.029,9-9s-4.029-9-9-9H9c-4.971,0-9,4.029-9,9v140 c0,4.971,4.029,9,9,9h140c4.971,0,9-4.029,9-9V86.201C158,81.23,153.971,77.201,149,77.201z"
                  />
                </g>
              </svg>
            </a>
            <a
              aria-label="Github"
              href="https://kelvinabella.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                role="img"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  GitHub
                </title>
                <path
                  d="M12 .297c-6.63 0-12 5.373-12 12 0 5.303 3.438 9.8 8.205 11.385.6.113.82-.258.82-.577 0-.285-.01-1.04-.015-2.04-3.338.724-4.042-1.61-4.042-1.61C4.422 18.07 3.633 17.7 3.633 17.7c-1.087-.744.084-.729.084-.729 1.205.084 1.838 1.236 1.838 1.236 1.07 1.835 2.809 1.305 3.495.998.108-.776.417-1.305.76-1.605-2.665-.3-5.466-1.332-5.466-5.93 0-1.31.465-2.38 1.235-3.22-.135-.303-.54-1.523.105-3.176 0 0 1.005-.322 3.3 1.23.96-.267 1.98-.399 3-.405 1.02.006 2.04.138 3 .405 2.28-1.552 3.285-1.23 3.285-1.23.645 1.653.24 2.873.12 3.176.765.84 1.23 1.91 1.23 3.22 0 4.61-2.805 5.625-5.475 5.92.42.36.81 1.096.81 2.22 0 1.606-.015 2.896-.015 3.286 0 .315.21.69.825.57C20.565 22.092 24 17.592 24 12.297c0-6.627-5.373-12-12-12"
                />
              </svg>
            </a>
          </span>
        </div>
        <div
          class="style__AppDetails-xfidxt-2 lfHxmc"
        >
          <h2>
            kelvinabella.com
          </h2>
          <span>
            react
          </span>
          <p>
            description
          </p>
        </div>
      </div>
    </div>
  </section>
  <footer
    class="style__FooterContainer-sc-1egeub5-0 jrWLre"
  >
    <div>
      <span>
        powered by 
        <a
          href="https://nextjs.org"
        >
          Next.js
        </a>
         and 
        <a
          href="https://reactjs.org"
        >
          React
        </a>
      </span>
      <span>
         | built by 
        <a
          href="https://kelvinabella.com"
        >
          Kelvin Abella
        </a>
         with ♥
      </span>
    </div>
  </footer>
</DocumentFragment>
`;
